python/dgl/distributed/optim/pytorch/sparse_optim.py

```diff
@@ -35,16 +46,194 @@ def __init__(self, params, lr):
             self._rank = 0
             self._world_size = 1
 
+    def local_state_dict(self):
+        """Return the state pertaining to current rank of the optimizer.
+
+        Returns
+        -------
+        dict
+            Local state dict
+
+        See Also
+        --------
+        load_local_state_dict
+        """
+        lcoal_state_dict = {}
+        lcoal_state_dict["emb_states"] = {}
+        num_clients = dgl.distributed.get_num_client()
+        lcoal_state_dict["params"] = {"num_clients": num_clients}
+        for emb in self._params:
+            kvstore = emb.kvstore
+            trainers_per_machine = (
+                num_clients // dgl.distributed.get_num_machines()
+            )
+            emb_state_dict = {}
+            part_policy = (
+                emb.part_policy if emb.part_policy else emb._tensor.part_policy
+            )
+            idx = self._get_local_ids(part_policy)
+            if trainers_per_machine > 1:
+                kv_idx_split = th.remainder(idx, trainers_per_machine).long()
```

Comment:
I believe you can also use `idx % trainers_per_machine`.  In fact you have it in L78.  Better make it consistent.

Reply:
fixed and add dict declaration in ```save``` because it is the API exposed to user.